### PR TITLE
build llvm 16.0 with riscv32 on linux

### PR DESCRIPTION
**Pull request by xermicus**

### CODE REVIEW

File: .github/workflows/build-llvm.yml [hunk 1]
```diff
@@ -6,6 +6,27 @@ on:
 name: Build LLVM Binaries
 
 jobs:
+  rv32e-linux-x86-64:
+    name: rv32e Linux x86-64
+    runs-on: ubuntu-20.04
+    steps:
```
Comment (seanyoung) [inline .github/workflows/build-llvm.yml:12]: what about the all the other platforms that solang supports?

File: build-rv32e.sh [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+
+cd llvm-project
+patch -p1 < ../patches/llvm-D70401.patch
+patch -p1 < ../patches/compiler-rt.patch
+
+
+mkdir -p build
+cd build
+cmake -G Ninja -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=$INSTALL_DIR -DLLVM_ENABLE_PROJECTS='clang;lld' -DLLVM_TARGETS_TO_BUILD="X86;AArch64;RISCV;WebAssembly" -DLLVM_ENABLE_RUNTIMES=compiler-rt ../llvm
```
Comment (seanyoung) [inline build-rv32e.sh:16]: Why X86 and Aarch64? That just bloats the libraries and therefore the solang binaries

File: .github/workflows/build-llvm.yml [hunk 3]
```diff
@@ -6,6 +6,27 @@ on:
...
+    - name: Checkout sources
+      uses: actions/checkout@v2
+      with:
+        submodules: recursive
+    - run: git clone --depth 1 --branch release/16.x https://github.com/llvm/llvm-project.git
```
Comment (seanyoung) [inline .github/workflows/build-llvm.yml:17]: So this means we're using llvm15 and llvm16 in solang. How can we switch between them?

File: .github/workflows/build-llvm.yml (hunk 1 above)
Comment (seanyoung) [reply .github/workflows/build-llvm.yml:12]: Also you've not added it to the ci image

**Comment by xermicus** [comment]
@seanyoung this is experimental and only used for my riscv branch

**Comment by seanyoung** [comment]
I've created some PRs to move to llvm 16. With that it should be possible to create a single llvm build with all the solana and riscv patches necessary, wdyt

**Comment by xermicus** [comment]
Ok we can try this. I don't know the solana labs LLVM fork, but chances are that it will just work. I'll build it locally to try it out first and let you know.

File: build-rv32e.sh (hunk 2 above)
Comment (xermicus) [reply build-rv32e.sh:16]: Good catch I removed them

File: .github/workflows/build-llvm.yml (hunk 3 above)
Comment (xermicus) [reply .github/workflows/build-llvm.yml:17]: Looks like we already switched, thanks!

File: .github/workflows/build-llvm.yml (hunk 1 above)
Comment (xermicus) [reply .github/workflows/build-llvm.yml:12]: > what about the all the other platforms that solang supports?

For now we only need Linux.. maybe ARM MacOS

**Pull request closed without merging**